modules/oe_starter_content_event/src/Plugin/Field/FieldFormatter/EventLinkExternalFormatter.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Drupal\oe_starter_content_event\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\link\Plugin\Field\FieldFormatter\LinkFormatter;
+
+/**
+ * Plugin implementation of the 'link_separate' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "event_link_external",
+ *   label = @Translation("Event links external opens in new tab"),
+ *   field_types = {
+ *     "link"
+ *   }
+ * )
+ */
+class EventLinkExternalFormatter extends LinkFormatter {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $element = parent::viewElements($items, $langcode);
+
+    foreach ($items as $delta => $item) {
+      $url = $this->buildUrl($item);
+
+      if ($url->isExternal()) {
+        $element[$delta]['#options']['attributes']['target'] = '_blank';
+      }
+    }
+
+    return $element;
+  }
+
```

Comment:
as a note, we could have just done a `{% if node.field_something.0.url.external %}` in the field template in whitelabel and skipped all this.
There already is a target blank setting in the link formatter, this is just a different behaviour.
Worth a formatter? I personally prefer formatters over templates, maybe we should have waited till another field would need this, I'm conflicted..

Reply:
we discussed to keep the formatter approach